src/common/lib/nostr.ts

```diff
@@ -17,6 +20,13 @@ const nostr = {
   hexToNip19(hex: string, prefix = "nsec") {
     return bech32Encode(prefix, hex);
   },
+  generatePublicKey(privateKey: string) {
```

Comment:
Generates public key from private key, rename to `derivePublicKey`.

Reply:
Use bg script action, review usages of this function.

Reply:
maybe the method can be moved to the nostr object (to review)

Reply:
I don't think this can be moved yet. In the nostr settings you can set a new private key and public key will be updated without saving (I think how this works could be changed. I will add a TODO)